internal/kafka/internal/services/kafka.go

```diff
@@ -1032,10 +1044,13 @@ func buildManagedKafkaCR(kafkaRequest *dbapi.KafkaRequest, kafkaConfig *config.K
 	if err != nil {
 		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "unable to list kafka request")
 	}
+
 	labels := map[string]string{
 		"bf2.org/kafkaInstanceProfileQuotaConsumed": strconv.Itoa(k.QuotaConsumed),
 		"bf2.org/kafkaInstanceProfileType":          kafkaRequest.InstanceType,
+		v1.ManagedKafkaBf2SuspendedLabelKey:         fmt.Sprintf("%t", arrays.Contains(constants.GetSuspendedStatuses(), kafkaRequest.Status)),
```

Comment:
@JameelB it's possible in the case where the control plane sets a version for Strimzi, Kafka, or Kafka IBP that is invalid in the data plane, or if the deployment is changed to reserved, the profile is changed (standard <-> developer), or the size changes (x1 <-> x2). Otherwise, the data plane overrides any other status with suspended when the label is present.

Reply:
I think the only valid next state from `suspending` from user point of view should be `suspended`. And the only valid state from `suspended` should be `resuming`.
ie. the CP in its state transition has to guard against these states changes `suspending -> failed`, `suspending->ready`, `suspended -> failed`, `suspended->ready`. 

and allow for `suspending->suspended` ......(allow for Kafka components version upgrades without states changes when they are needed)...  `suspended->resuming`,  ( either `resuming->failed` or `resuming->ready`) states.

or allow for `suspending->suspended` ......(allow for Kafka components version upgrades without states changes when they are needed)...  `suspended->deprovision`... kafka deletion.

Reply:
Thanks @MikeEdgar @machi1990. 

I was thinking the same, but we won't see if a Kafka fails to go from `suspending` -> `suspended`. I think what we're missing here are metrics + alert if a Kafka is stuck in `suspending` for quite some time. wdyt?

This can be done as part of a separate JIRA.

Reply:
> I was thinking the same, but we won't see if a Kafka fails to go from suspending -> suspended. I think what we're missing here are metrics + alert if a Kafka is stuck in suspending for quite some time. 

I think there will be sentry alerts (https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/1007c644febe368beacedbbe0cbb5f03b01027b2/internal/kafka/internal/services/data_plane_kafka.go#L311) when a Data Plane reports that the Kafka has failed. We could add a metric on top of that.